bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -845,6 +854,30 @@ protected void release(boolean wait)
       }
     }
 
+    private boolean canRelease(boolean writer)
```

Comment:
I think the caller makes the decision to do the spooling. This method is just saying the block can be released because it is not needed so I am thinking the general thought around the method name is fine.